### PR TITLE
Make METS adapter mark bags with no assets except the METS file as deleted

### PR DESCRIPTION
**Pull request by alicefuzier**
For https://github.com/wellcomecollection/platform/issues/4893
This PR only changes the behaviour of the METS adapter and will require a reharvest. A subsequent PR will change the trasformer to explicitly mark deleted works

### CODE REVIEW

File: pipeline/transformer/transformer_mets/src/test/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsXmlTransformerTest.scala [hunk 1]
```diff
@@ -30,6 +30,11 @@ class MetsXmlTransformerTest
     )
   }
 
+  it("should transform blup") {
```
Comment (kenoir) [inline pipeline/transformer/transformer_mets/src/test/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsXmlTransformerTest.scala:33]: what is blup?

**Review by kenoir — APPROVED** [review]

File: pipeline/transformer/transformer_mets/src/test/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsXmlTransformerTest.scala (hunk 1 above)
Comment (alicefuzier) [reply pipeline/transformer/transformer_mets/src/test/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsXmlTransformerTest.scala:33]: ooops sorry

File: mets_adapter/mets_adapter/src/main/scala/uk/ac/wellcome/mets_adapter/models/Bag.scala [hunk 2]
```diff
@@ -10,17 +10,19 @@ case class Bag(info: BagInfo,
...
-            file,
-            createdDate,
-            manifestations)
+          MetsSourceData(
+            bucket = location.bucket,
+            path = location.path,
+            version = version,
+            file = file,
+            createdDate = createdDate,
+            deleted = manifest.files.forall(f => f.path == file),
```
Comment (alexwlchan) [inline mets_adapter/mets_adapter/src/main/scala/uk/ac/wellcome/mets_adapter/models/Bag.scala:23]: Might be worth a comment here explaining that "only the METS file" => deleted, with a link to the ticket/discussion.

**Review by alexwlchan — APPROVED** [review]
👍

Looks good. Thanks for clarifying the scope in the PR description.

**Pull request merged**